### DISCUSSION TITLE
Beautify Textadept with Clearlooks engine

### DISCUSSION THREAD

**Discussion by brynne8** [orbitalquark/textadept#18] [Ideas]
Previously, in Windows, Textadept used a GTK engine called "wimp", it's outdated and lack of documentation. Today, I accidentally discovered a better engine, Clearlooks, which is able to present a modern UI. I found it at https://www.gimp-forum.net/Thread-Stylizing-GtkNotebook-Tabs-in-Theme-gtkrc. It not only improves the scrollbar look, but also the menu, and overall experience. Here are screenshots on Windows 7 (with [Gabriel's ta-tweaks](https://github.com/gabdub/ta-tweaks) installed).
![screenshot](https://user-images.githubusercontent.com/7542439/103187070-efadc280-48fd-11eb-917d-7f2a9c0d7c5a.png)
![screenshot](https://user-images.githubusercontent.com/7542439/103187181-6480fc80-48fe-11eb-97a7-5af436175bc8.png)

---
PS: In Windows 10, one line has to be commented out in gtkrc:
```
style "scrollbar" {
  ...
  #bg[ACTIVE] = @hl_color
  ...
}
```

**Comment by orbitalquark** [comment]
I think I tried Clearlooks long ago on Windows, but Textadept did not look like other Windows applications using that theme. You showed some pictures with a "dark" theme, and after briefly scrolling through the forum you linked to, the GIMP screenshots also use a dark theme. How does Textadept look with a "light" theme? Does it look like other Windows applications? Or are there differences? I'd prefer to keep Textadept looking as Windows-native as possible.

**Comment by sigzero** [comment]
The tab shape on that image is definitely not "Windows" style. However, it would be nice to get a better experience overall if possible. Clearlooks works with GTK2 and GTK3 with the Clearlooks-Phenix or something (assuming TA is GTK2 though).

It would be nice if more screenshots could be posted (with dark and light themes) showing the various UI elements. This would definitely be a good experiment.

Note: This should be done without the ta-tweaks as "standard" Textadept needs to be seen.

**Comment by sigzero** [comment]
Anything more about updating to a newer version? Or is this a dead issue?

**Comment by orbitalquark** [comment]
Oh, sorry for not saying something more definitive. I mentioned previously that I prefer Textadept look as Windows-native as possible, and you replied that the Clearlooks theme definitely does not look like Windows, so to me it felt like there was no point switching. Since you prefer Clearlooks, have you tried installing it in *lib/gtk-2.0/2.10.0/engines* and *share/themes*, and then perhaps editing *share/themes/MS-Windows/gtk-2.0/gtkrc* to use Clearlooks? I'm not actually sure how to change the gtk theme for a GTK on Windows application :(

**Reply by sigzero** [reply]
Sadly, I am just a user and not a developer. However, I think a more modern engine will get you a much better LAF.

**Comment by sigzero** [comment]
Maybe look at how Scite does it? It also uses Gtk and Scintilla but its dialogs look Windows native and I believe native on the platforms it supports where Textadept does not. Just a thought.

**Comment by orbitalquark** [comment]
SciTE uses native Windows widgets. There is no GTK. Geany uses GTK on Windows, but I haven't run it for comparison.

**Reply by sigzero** [reply]
Ah, what a shame.
